packages/@aws-cdk/toolkit-lib/test/_helpers/mock-sdk.ts

```diff
@@ -144,14 +144,23 @@ export const setDefaultSTSMocks = () => {
  * clients may be used.
  */
 export class MockSdkProvider extends SdkProvider {
+  private defaultAccounts: string[] = [];
+
   constructor() {
     super(FAKE_CREDENTIAL_CHAIN, 'bermuda-triangle-1337', {
       ioHelper: new TestIoHost().asHelper('sdk'),
     });
   }
 
+  public returnsDefaultAccounts(...accounts: string[]) {
+    this.defaultAccounts = accounts;
+  }
+
   public defaultAccount(): Promise<Account | undefined> {
-    return Promise.resolve({ accountId: '123456789012', partition: 'aws' });
+    const accountId = this.defaultAccounts.length === 0
+      ? '123456789012'
+      : this.defaultAccounts.shift()!;
```

Comment:
sure but why not `const accountId = this.defaultAccounts.shift() ?? '123456789012';`

Reply:
Cause I just copied it!

Reply:
well whoever wrote that seems to not know that the nullish coalescer exists...